### PR TITLE
Add TestableMessageSession and TestableEndpointInstance

### PR DESCRIPTION
**Pull request by timbussmann**
testable implementation for `IMessageSession` and `IEndpointInstance` (implementing `IMessageSession`) aligned with the implementation for `TestableMessageHandlerContext`

Connects to Particular/NServiceBus.Testing#29

### CODE REVIEW

**Comment by timbussmann** [comment]
@Particular/nservicebus-maintainers @mat-mcloughlin @WojcikMike ping

File: src/NServiceBus.Core.Tests/Fakes/TestableMessageSessionTests.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+            session.Subscribe(typeof(MyEvent), options);
+
+            Assert.AreEqual(1, session.Subscriptions.Length);
+            Assert.AreSame(options, session.Subscriptions[0].Options);
+            Assert.AreEqual(typeof(MyEvent), session.Subscriptions[0].Message);
+        }
+
+        [Test]
+        public void Unsubscribe_ShouldTrackUnsubscribpions()
```
Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/Fakes/TestableMessageSessionTests.cs]: Spelling

**Comment by danielmarbach** [comment]
One spelling, apart from that :+1:

**Comment by timbussmann** [comment]
@danielmarbach fixed

**Comment by andreasohlund** [comment]
Should we include the fakes in the apiapprovals? (can be a new pull)

**Comment by danielmarbach** [comment]
@andreasohlund they are approved in the testing package. Enough?

**Comment by andreasohlund** [comment]
> they are approved in the testing package. Enough?

Yea, since we actually release them in binary form from the testing repo that makes more sense. (ie the .source package is for intenal use only?)

**Comment by danielmarbach** [comment]
yes

**Pull request merged**